Vaults/Nodsoft.MoltenObsidian.Vaults.Http/VaultDependencyInjectionExtensions.cs

```diff
@@ -44,4 +39,21 @@ public static IServiceCollection AddMoltenObsidianHttpVault(this IServiceCollect
 	/// <returns>The service collection.</returns>
 	public static IServiceCollection AddMoltenObsidianHttpVault(this IServiceCollection serviceCollection, string vaultRootUri)
 		=> serviceCollection.AddMoltenObsidianHttpVault(_ => new() { BaseAddress = new(vaultRootUri) });
+}
+
+public static class AsyncVaultDependencyInjectionExtensions
+{
+	public static IServiceCollection AddMoltenObsidianHttpAsyncVault(this IServiceCollection serviceCollection, Func<IServiceProvider, HttpClient> httpClientProvider)
+		=> serviceCollection.AddSingleton<Task<IVault>>(async services =>
+		{
+			HttpClient httpClient = httpClientProvider(services);
+
+			// Get the vault manifest from the server
+			RemoteVaultManifest manifest = await httpClient.GetFromJsonAsync<RemoteVaultManifest>("moltenobsidian.manifest.json")
+				?? throw new InvalidOperationException("Failed to retrieve the vault manifest from the server.");
+
+			return HttpRemoteVault.FromManifest(manifest, httpClient);
+		});
+	public static IServiceCollection AddMoltenObsidianHttpAsyncVault(this IServiceCollection serviceCollection, string vaultRootUri)
+		=> serviceCollection.AddMoltenObsidianHttpAsyncVault(_ => new() { BaseAddress = new(vaultRootUri) });
```

Comment:
Please add XML comments to document method signature. Bonus points if you can tell which to use in which circumstances.